### PR TITLE
Add support for dedicated network per packet

### PR DESCRIPTION
**Pull request by bergie**

### CODE REVIEW

File: src/main.coffee [hunk 1]
```diff
@@ -17,6 +17,7 @@ parse = (args) ->
     .option('--attr key.subkey=value', 'Additional attributes', addOption, [])
     .option('--trace [true]', 'Enable tracing with Flowtrace', Boolean, false)
     .option('--cache [true]', 'Enable NoFlo component cache', Boolean, false)
+    .option('--dedicated_network [true]', 'Create a dedicated NoFlo network for each packet', Boolean, false)
```
Comment (jonnor) [inline src/main.coffee]: Should be `--dedicated-network`. The option will then be available as `options.dedicatedNetwork`

File: src/mount.coffee [hunk 2]
```diff
@@ -112,20 +157,28 @@ loadAndStartGraph = (loader, graphName, iips, callback) ->
...
-        # needs to happen after NoFlo network has sent its IIPs
-        debug 'sending IIPs', Object.keys(iips)
-        sendIIPs instance, iips
-        return callback null, instance
-      setTimeout onStarted, 100 # XXX: hack, remove when https://github.com/noflo/noflo/issues/261 fixed
+
+      # Components don't have a start callback, we can just go started immediately
+      setTimeout ->
+        do onStarted
+      , 100
```
Comment (jonnor) [inline src/mount.coffee]: Please add a `# FIXME: without this some tracing groups are missing?`

File: src/mount.coffee [hunk 3]
```diff
@@ -290,7 +348,7 @@ class Mounter
     # Handle trace subprotocol
     switch command
       when 'start'
-        @tracer.attach @instance.network
+        @tracer.attach @instances[0].network
```
Comment (jonnor) [inline src/mount.coffee]: This is broken. Should loop through the instances and add the network of each one

File: .travis.yml [hunk 4]
```diff
@@ -17,6 +17,8 @@ install:
 before_script:
 - mosquitto -d
 - npm install -g grunt-cli
+#env:
```
Comment (jonnor) [inline .travis.yml]: Dead code, please remove

File: spec/mount.coffee [hunk 5]
```diff
@@ -282,16 +291,15 @@ transportTest = (address) ->
...
+      transportTest options
 
     describe 'attempting stop without start', () ->
       it 'should return sucess without doing anything', (done) ->
-        options =
-          broker: address
+        newOptions = extendOptions options,
           graph: 'core/RepeatAsync'
           name: '1someone'+randomstring.generate 4
         m = new mount.Mounter options
```
Comment (jonnor) [inline spec/mount.coffee]: You create `newOptions` but this still takes `options`, is wrong?

File: src/mount.coffee [hunk 6]
```diff
@@ -57,11 +56,18 @@ wrapOutport = (transactions, client, instance, port, queueName) ->
         fields:
           deliveryTag: group
       data: null
+
+    result =
+      id: group
```
Comment (jonnor) [inline src/mount.coffee]: If `result` should have `.type`, then it should be initialized here, so this is clear

File: src/mount.coffee [hunk 7]
```diff
@@ -78,6 +84,45 @@ wrapOutport = (transactions, client, instance, port, queueName) ->
...
+
+# Execute each packet using a dedicated NoFlo network
+wrapPortsDedicated = (transactions, client, loader, definition, options) ->
+  definition.inports.forEach (port) ->
+    return unless port.queue
+    return unless port.id
+
+    onMessage = (msg) ->
+      loadAndStartGraph loader, options.graph, options.iips, (err, instance) ->
+        if err
```
Comment (jonnor) [inline src/mount.coffee]: This sounds like a pretty bad error... Should we just throw so it will be fatal?

**Comment by bergie** [comment]
@jonnor those comments should be addressed now

**Pull request merged**